src/sage/combinat/multiset_partition_into_sets_ordered.py

```diff
@@ -3399,7 +3400,7 @@ def val(self, q='q'):
             (q^2+q+1)*s[2, 1, 1] + q*s[2, 2]
         """
         H = [self._OMPs(list(b)) for b in self.highest_weight_vectors()]
-        Sym = SymmetricFunctions(ZZ[q])
+        Sym = SymmetricFunctions(PolynomialRing(ZZ, q))
```

Comment:
So, for example, this one?  Only for `ZZ`, or also when the base ring *could be* `ZZ`?

Maybe we should revert all cases where it is not clear that the argument must be a string?

Reply:
To be safe, all cases where the argument is not guaranteed to be a string. That way we don't introduce any backwards-incompatible changes.